### PR TITLE
[Improve][SourceConnector] Unifie Iceberg source fields to schema

### LINKED ISSUE #4113: [Improve][SourceConnector]  update Iceberg source fields to schema, unified connector schema parameter
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Unified schema parameter, update Iceberg source option **fields** to **schema**.

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by wfrong**
fix  https://github.com/apache/incubator-seatunnel/issues/4113

Unifie Iceberg source fields to schema

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by CalvinKirs** [comment]
Do we need to do a compatibility adaptation? (it doesn't have to be in every connector)

**Comment by EricJoy2048** [comment]
> Do we need to do a compatibility adaptation? (it doesn't have to be in every connector)

I agree with you.

**Comment by wfrong** [comment]
> Do we need to do a compatibility adaptation? (it doesn't have to be in every connector)

Thanks for your suggestions！I have added compatibility adaptation. Could you help to review again?

**Comment by wfrong** [comment]
cc @CalvinKirs @liugddx  could you help me review this PR?

**Review by EricJoy2048 — DISMISSED** [review]

**Comment by EricJoy2048** [comment]
LGTM

File: seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/source/IcebergSource.java [hunk 1]
```diff
@@ -105,9 +105,15 @@ private SeaTunnelRowType loadSeaTunnelRowType(Schema tableSchema, Config pluginC
...
-                CheckConfigUtil.checkAllExists(pluginConfig, CommonConfig.KEY_FIELDS.key());
+                CheckConfigUtil.checkAtLeastOneExists(
+                        pluginConfig, CommonConfig.KEY_FIELDS.key(), SeaTunnelSchema.SCHEMA.key());
+
         if (checkResult.isSuccess()) {
-            SeaTunnelSchema configSchema = SeaTunnelSchema.buildWithConfig(pluginConfig);
+            Config config =
+                    pluginConfig.hasPath(SeaTunnelSchema.SCHEMA.key())
+                            ? pluginConfig.getConfig(SeaTunnelSchema.SCHEMA.key())
+                            : pluginConfig;
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/source/IcebergSource.java:115]: Do you mean `pluginConfig.getConfig(CommonConfig.KEY_FIELDS.key())`?

**Review by TyrantLucifer — CHANGES_REQUESTED** [review]

File: seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/source/IcebergSource.java (hunk 1 above)
Comment (wfrong) [reply seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/source/IcebergSource.java]: Yes, thank you very much! I will fix this.

Comment (wfrong) [reply seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/source/IcebergSource.java:115]: I mean if user use  `schema`  in config file , we use `pluginConfig.getConfig(SeaTunnelSchema.SCHEMA.key())`.  And to support some old tasks which only use `fields` we just use orginal config.

Comment (wfrong) [reply seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/source/IcebergSource.java:115]: > Do you mean `pluginConfig.getConfig(CommonConfig.KEY_FIELDS.key())`?

I mean if user use  `schema`  in config file , we use `pluginConfig.getConfig(SeaTunnelSchema.SCHEMA.key())`.  And to support some old tasks which only use `fields` we just use orginal config.

Comment (TyrantLucifer) [reply seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/source/IcebergSource.java:115]: Got it.

**Review by TyrantLucifer — APPROVED** [review]

**Comment by wfrong** [comment]
> LGTM

could you help me to review this PR again?

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**